### ISSUE TITLE
Reason for cloning the images and labels ?

### ISSUE THREAD

**Issue by kfarivar** [Harry24k/adversarial-attacks-pytorch#57]
Hello,
thank you for the library. 

I was wondering why you clone and detach the images as the first step of many attacks? 
cloning doubles the amount of memory used in GPU which reduces speed.
wouldn't zeroing the gradient for the images be sufficient?

**Comment by Harry24k** [comment]
As your concern, cloning doubles the amount of memory.
However, without cloning, the original gradient information of images should be cleaned for the attack process.
This means that it leads to unpredictable errors in some processes:
``
atk = torchattacks.PGD(eps=8/255, ...)
images.requires_grad = True
adv_images = atk(images, labels)
loss(images, adv_images).backward()
``
Here, `images' has the following value as a gradient: (the last gradient of the attack process) + (the gradient of backward).
However, the code seems like `images' should have the gradient of backward as a gradient.
In summary, it sacrifices some memory for intuitive code.
Additionally, the memory seized by images usually less than the memory seized by the model parameters.
